src/openqaoa-core/backends/wrapper.py

```diff
@@ -0,0 +1,200 @@
+            calibration_measurements,
+            calibration_registers,
+            qubit_mapping,
+        )
+
+        """
+        This doesn't work because keys are tuples  =(
+        # save calibration factors just in case
+        with open('test.json', "w") as fp:
+            json.dump(self.calibration_factors, fp)
+        """
+
+    def get_counts(self, params, n_shots=None, seed=42):
+        """
+        Overrides the get_counts function of the backend object.
+        Modified function to...
+            divide into batches
+            change the self.append_state according to the schedule, s.
+            get the counts and classically negate them
+            combine all batches into a count dict under BFA
+        """
+        random.seed(seed)
+        # list of integers whose binary representation signifies which qubits to be flipped at every batch
+        s_list = []
+        for _ in range(0, self.n_batches):
+            s_list.append(random.getrandbits(self.backend.n_qubits))
+
+        n_shots = self.backend.n_shots if n_shots == None else n_shots
+        n_shots_batch = int(n_shots / self.n_batches)
+
+        counts = {}
+
+        for batch in range(0, self.n_batches):
+            s = s_list[batch]
+            s_binary = format(s, "b").zfill(self.backend.n_qubits)  # convert to binary
+            arr = np.fromiter(s_binary, dtype=int)
+            negated_qubits = np.where(arr == 1)[0]  # where the syndrome has a 1
+
+            circuit_to_append = self.backend.gate_applicator.create_quantum_circuit(
+                self.backend.n_qubits
+            )
+
+            for negated_qubit in negated_qubits:
+                negated_qubit = negated_qubit.item()
+                negation_gate = X(self.backend.gate_applicator, negated_qubit)
+                circuit_to_append = self.backend.gate_applicator.apply_gate(
+                    negation_gate, negated_qubit, circuit_to_append
+                )
+
+            self.backend.append_state = circuit_to_append
+
+            counts_batch = self.backend.get_counts(
+                params, n_shots_batch
+            )  # calls the get_counts method of the specific backend
+
+            negated_counts = negate_counts_dictionary(
+                counts_dictionary=counts_batch, s=s
+            )
+
+            # Add to the final counts dict
+            for key in negated_counts:
+                if key in counts:
+                    counts[key] = counts[key] + negated_counts[key]
+                else:
+                    counts.update([(key, negated_counts[key])])
+
+        self.measurement_outcomes = counts
+
+        return counts
+
+    @round_value
+    def expectation_value_spam_twirled(
+        self, counts: Dict, hamiltonian: Hamiltonian, calibration_factors: dict
+    ):
+        """
+        TODO
+        """
+
+        terms = [term.qubit_indices for term in hamiltonian.terms]
+        hamiltonian_as_dict = hamiltonian.hamiltonian_dict()
```

Comment:
No need to create an extra variable. Just pass `hamiltonian.hamiltonian_dict()` as the argument in lines 169 and 177

Reply:
Are you sure that this won't slow the code? Because it will be calling this function often, it's inside a for loop over all terms of the hamiltonian.

Reply:
Good point: try!

Time it and report which one is faster. You can use https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-timeit